contrib/golang/filters/http/source/go/pkg/http/filtermanager.go

```diff
@@ -63,3 +65,254 @@ func getHttpFilterConfigParser(name string) api.StreamFilterConfigParser {
+		return nil, err
+	}
+
+	protos := configStruct.GetConfigs()
+	for _, proto := range protos {
+		name := proto.Name
+		if v, ok := httpFilterConfigFactoryAndParser.Load(name); ok {
+			plugin := v.(*filterConfigFactoryAndParser)
+			config, err := plugin.configParser.Parse(proto.Config, nil)
+			if err != nil {
+				return nil, fmt.Errorf("%w during parsing plugin %s in filtermanager", err, name)
+			}
+
+			conf.current = append(conf.current, &filterConfig{
+				Name:         proto.Name,
+				parsedConfig: config,
+			})
+		} else {
+			api.LogErrorf("plugin %s not found, ignored", name)
+		}
+	}
+
+	return conf, nil
+}
+
+func (p *filterManagerConfigParser) Merge(parent interface{}, child interface{}) interface{} {
+	childConfig := child.(*filterManagerConfig)
+
+	// TODO: We have considered to implemented a Merge Policy between the LDS's filter & RDS's per route
+	// config. A thought is to reuse the current Merge method. For example, considering we have
+	// LDS:
+	//	 - name: A
+	//	   pet: cat
+	// RDS:
+	//	 - name: A
+	//	   pet: dog
+	// we will call plugin A's Merge method, which will produce `pet: [cat, dog]` or `pet: dog`.
+	// As there is no real world use case for the Merge feature, I decide to delay its implementation
+	// to avoid premature design.
+	newConfig := *childConfig
+	return &newConfig
+}
+
+type filterManager struct {
+	filters []api.StreamFilter
+
+	callbacks api.FilterCallbackHandler
+}
+
+func filterManagerConfigFactory(c interface{}) api.StreamFilterFactory {
+	conf, ok := c.(*filterManagerConfig)
+	if !ok {
+		panic("unexpected config type")
+	}
+
+	newConfig := conf.current
+	factories := make([]api.StreamFilterFactory, len(newConfig))
+	for i, fc := range newConfig {
+		var factory api.StreamFilterConfigFactory
+		name := fc.Name
+		if v, ok := httpFilterConfigFactoryAndParser.Load(name); ok {
+			plugin := v.(*filterConfigFactoryAndParser)
+			factory = plugin.configFactory
+			config := fc.parsedConfig
+			factories[i] = factory(config)
+
+		} else {
+			api.LogErrorf("plugin %s not found, pass through by default", name)
+			factory = PassThroughFactory
+			factories[i] = factory(nil)
+		}
+	}
+
+	return func(callbacks api.FilterCallbackHandler) api.StreamFilter {
+		filters := make([]api.StreamFilter, len(factories))
+		for i, factory := range factories {
+			filters[i] = factory(callbacks)
+		}
+		return &filterManager{
+			callbacks: callbacks,
+			filters:   filters,
+		}
+	}
+}
+
+func (m *filterManager) DecodeHeaders(header api.RequestHeaderMap, endStream bool) api.StatusType {
+	go func() {
+		defer m.callbacks.RecoverPanic()
+
+		for _, f := range m.filters {
+			// When the filter is controlled by filterManager, it don't
+			// need to create goroutine by itself. So there is no need
+			// to return api.Running and resume inside the filter. We
+			// treat api.Running as api.Continue here.
+			status := f.DecodeHeaders(header, endStream)
+			if status != api.Running && status != api.Continue {
```

Comment:
seems `api.Running` is meaningless for a sub-filter?
Also, maybe `api.Stop` is meaningless too?

Reply:
> seems api.Running is meaningless for a sub-filter?

Yes

> Also, maybe api.Stop is meaningless too?

Stop can be used to skip the later filter in that phase